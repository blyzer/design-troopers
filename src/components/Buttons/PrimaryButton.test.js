import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { PrimaryButton } from './index.js';

describe('<PrimaryButton /> works', () => {
  test('it renders properly', () => {
    const { getByRole } = render(<PrimaryButton>Sign Up</PrimaryButton>);
    expect(getByRole('button')).toMatchSnapshot();
  });

  test('on click button triggers callback', () => {
    const spyButtonClick = jest.fn();
    const { getByRole } = render(<PrimaryButton onClick={spyButtonClick}>Sign Up</PrimaryButton>);
    fireEvent.click(getByRole('button'));
    expect(spyButtonClick).toHaveBeenCalled();
  });
});
