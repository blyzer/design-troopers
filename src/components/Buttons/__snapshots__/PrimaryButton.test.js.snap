// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton /> works it renders properly 1`] = `
<button
  class="dt-bg-primary-500 dt-text-white hover:dt-bg-primary-600 focus:dt-bg-primary-700 dt-py-2 dt-px-6 dt-rounded dt-shadow-lg dt-border dt-border-gray-300 dt-flex dt-items-center dt-justify-center dt-transition-all dt-transition-500 focus:dt-outline-none"
  data-name="DTPrimaryButton"
>
  Sign Up
</button>
`;
